addon/lib/StudyTelemetryCollector.jsm

```diff
@@ -33,47 +32,50 @@ XPCOMUtils.defineLazyModuleGetter(this, "PlacesUtils",
 XPCOMUtils.defineLazyServiceGetter(this, "Telemetry",
   "@mozilla.org/base/telemetry;1", "nsITelemetry");
 
+// unit-tested study helpers
+XPCOMUtils.defineLazyModuleGetter(
+  this, "Helpers", "resource://esper-pioneer-shield-study/lib/Helpers.jsm"
+);
+
 /**
  * Note: Setting studyUtils in this constructor due to above comments
  */
 class StudyTelemetryCollector {
 
-  constructor(studyUtils, variation) {
-    this.variation = variation;
-    this.studyUtils = studyUtils;
+  constructor(pioneerUtils) {
+    this.pioneerUtils = pioneerUtils;
   }
 
   async start() {
 
-    this.telemetry({ event: "esper-init" });
+    console.log('ESPER Study init. Will now await full telemetry initialization before collecting the payload for this study.');
 
     // Ensure that we collect telemetry payloads only after it is fully initialized
     // See http://searchfox.org/mozilla-central/rev/423b2522c48e1d654e30ffc337164d677f934ec3/toolkit/components/telemetry/TelemetryController.jsm#295
-    TelemetryController.promiseInitialized().then(() => {
-
-      try {
+    await TelemetryController.promiseInitialized();
 
-        this.collectAndSendTelemetry();
+    try {
 
-      } catch (ex) {
-        // TODO: how are errors during study execution reported?
-        // this.studyUtils.telemetryError();
-        Components.utils.reportError(ex);
-      }
+      this.collectAndSendTelemetry();
 
-    });
+    } catch (ex) {
+      // TODO: how are errors during study execution reported?
+      // this.pioneerUtils.telemetryError();
+      Components.utils.reportError(ex);
+    }
 
   }
 
-  telemetry(payload) {
-    if (StudyTelemetryCollector.allowedToSendTelemetry()) {
-      this.studyUtils.telemetry(payload);
+  async telemetry(schemaName, schemaVersion, payload) {
+    if (await this.allowedToSendTelemetry()) {
```

Comment:
you don't need this check it is redundant, since it is already done in pioneer-utils.

if you want to have it just for logging you can rewrite this to:
```js
const pingId = await Pioneer.utils.submitEncryptedPing(schemaName, schemaVersion, payload);
if (pingId) {
  console.log('Ping sent.');
} else {
  console.log('Ping not sent.');
}
```

Reply:
As a side note... Does `console.log` actually work here? I've had trouble with getting `console.log` to work anywhere other than in the `startup` and `shutdown` methods of `bootstrap.js`.

If this is working as expected, great! If not, you may be better off using `Pioneer.utils.log.debug`:
```js
Pioneer.utils.log.debug('message goes here');
```

Reply:
Thanks, rewritten as per your suggestion. Console log works as expected btw :)